ANL/Polaris/spack.yaml

```diff
@@ -5,16 +5,16 @@
 spack:
   # add package specs to the `specs` list
   specs:
-  - mochi-margo^json-c@0.13.0
```

Comment:
We still need a way to pin the environment to json-c@0.13.0 specifically somehow (there may be a better way than the way this was done previously).  If it builds the newest json-c, then the resulting libraries will produce this link warning when linked against cray-mpich libraries:

```
  CCLD     perf-regression/ssg-test-separate-group-attach
/usr/lib64/gcc/x86_64-suse-linux/12/../../../../x86_64-suse-linux/bin/ld: warning: libjson-c.so.3, needed by /opt/cray/libfabric/1.15.2.0/lib64/libfabric.so.1, may conflict with libjson-c.so.5
```
(The problem is that cray-mpich depends on its own libfabric build, which in turn depends on its own libjson-c; we have to match it to avoid link warnings and possible abi compatibility problems)

Reply:
Ok, I had meant to ask you why this was necessary. My environment built fine without it but I haven't tried to run anything yet. The correct fix would be another "require":
```
packages:
  ...
  json-c:
    require: "@0.13.0"
```

Reply:
I just added it.

Reply:
perfect, thanks!